sqlglot/dialects/redshift.py

```diff
@@ -0,0 +1,58 @@
+from sqlglot import exp
+from sqlglot.dialects.postgres import Postgres
+from sqlglot.tokens import TokenType
+
+
+class Redshift(Postgres):
+    time_format = "'YYYY-MM-DD HH:MI:SS'"
+    time_mapping = {
+        "YY": "%y",
+        "yy": "%y",
+        "YYYY": "%Y",
+        "yyyy": "%Y",
+        "MM": "%m",
+        "mm": "%m",
+        "MON": "%b",
+        "mon": "%b",
+        "DD": "%d",
+        "dd": "%d",
+        "HH12": "%I",
+        "hh12": "%I",
+        "HH24": "%H",
+        "hh24": "%H",
+        "HH": "%H",
+        "hh": "%H",
+        "MI": "%M",
+        "mi": "%M",
+        "SS": "%S",
+        "ss": "%S",
+        "OF": "%z",
+        "of": "%z",
+        "AM": "%p",
+        "PM": "%p",
+    }
+
+    class Tokenizer(Postgres.Tokenizer):
+        ESCAPE = "\\"
+
+        KEYWORDS = {
+            **Postgres.Tokenizer.KEYWORDS,
+            "GEOMETRY": TokenType.GEOMETRY,
+            "GEOGRAPHY": TokenType.GEOGRAPHY,
+            "HLLSKETCH": TokenType.HLLSKETCH,
+            "SUPER": TokenType.SUPER,
+            "TIME": TokenType.TIMESTAMP,
+            "TIMETZ": TokenType.TIMESTAMPTZ,
+            "VARBYTE": TokenType.VARBYTE,
+            "SIMILAR TO": TokenType.SIMILAR_TO,
+        }
+
+    class Generator(Postgres.Generator):
+        TYPE_MAPPING = {
+            **Postgres.Generator.TYPE_MAPPING,
+            exp.DataType.Type.BINARY: "VARBYTE",
+            exp.DataType.Type.GEOMETRY: "GEOMETRY",
```

Comment:
do geometry and hll sketch, or super needs these mappings? doesn't seem to do anything

Reply:
I think there are some use cases. For example:
- https://hevodata.com/learn/redshift-super-data-type/  (`CREATE TABLE region_nations_noshred (rdata SUPER);`)
- https://www.youtube.com/watch?v=f4EpjeuOchE (`CREATE TABLE ... (... active_users HLLSKETCH, ...)`
- https://docs.aws.amazon.com/redshift/latest/dg/GeometryType-function.html (not explicitly used).

But I also didn't find many other references, so I could leave them out for the time being.